packages/graphql-transformers-e2e-tests/src/__tests__/FunctionTransformerTests.e2e.test.ts

```diff
@@ -0,0 +1,222 @@
+    return (outputs: Output[]) => {
+        const output = outputs.find((o: Output) => o.OutputKey === key)
+        return output ? output.OutputValue : null
+    }
+}
+
+beforeAll(async () => {
+    const validSchema = `
+    type Query {
+        echo(msg: String!): Context @function(name: "e2e-tests-echo-dev-${BUILD_TIMESTAMP}")
+        echoEnv(msg: String!): Context @function(name: "e2e-tests-echo-\${env}-${BUILD_TIMESTAMP}")
+        duplicate(msg: String!): Context @function(name: "e2e-tests-echo-dev-${BUILD_TIMESTAMP}")
+        pipeline(msg: String!): String
+            @function(name: "${ECHO_FUNCTION_NAME}")
+            @function(name: "${HELLO_FUNCTION_NAME}")
+        pipelineReverse(msg: String!): Context
+            @function(name: "${HELLO_FUNCTION_NAME}")
+            @function(name: "${ECHO_FUNCTION_NAME}")
+    }
+    type Context {
+        arguments: Arguments
+        typeName: String
+        fieldName: String
+    }
+    type Arguments {
+        msg: String!
+    }
+    `
+    try {
+        await awsS3Client.createBucket({Bucket: BUCKET_NAME}).promise()
+    } catch (e) { console.warn(`Could not create bucket: ${e}`) }
+    try {
+        const role = await IAM_HELPER.createLambdaExecutionRole(LAMBDA_EXECUTION_ROLE_NAME);
+        await wait(5000);
+        const policy = await IAM_HELPER.createLambdaExecutionPolicy(LAMBDA_EXECUTION_POLICY_NAME);
+        await wait(5000);
+        LAMBDA_EXECUTION_POLICY_ARN = policy.Policy.Arn;
+        await IAM_HELPER.attachLambdaExecutionPolicy(policy.Policy.Arn, role.Role.RoleName)
+        await wait(10000);
+        await LAMBDA_HELPER.createFunction(ECHO_FUNCTION_NAME, role.Role.Arn, 'echoFunction');
+        await LAMBDA_HELPER.createFunction(HELLO_FUNCTION_NAME, role.Role.Arn, 'hello');
+    } catch (e) { console.warn(`Could not setup function: ${e}`) }
+    const transformer = new GraphQLTransform({
+        transformers: [
+            new ModelTransformer(),
+            new FunctionTransformer()
+        ]
+    })
+    const out = transformer.transform(validSchema);
+    const finishedStack = await deploy(
+        customS3Client, cf, STACK_NAME, out, { env: 'dev' }, LOCAL_FS_BUILD_DIR, BUCKET_NAME, S3_ROOT_DIR_KEY,
+        BUILD_TIMESTAMP
+    )
+    // Arbitrary wait to make sure everything is ready.
+    await cf.wait(5, () => Promise.resolve())
+    console.log('Successfully created stack ' + STACK_NAME)
+    console.log(finishedStack)
+    expect(finishedStack).toBeDefined()
+    const getApiEndpoint = outputValueSelector(ResourceConstants.OUTPUTS.GraphQLAPIEndpointOutput)
+    const getApiKey = outputValueSelector(ResourceConstants.OUTPUTS.GraphQLAPIApiKeyOutput)
+    const endpoint = getApiEndpoint(finishedStack.Outputs)
+    const apiKey = getApiKey(finishedStack.Outputs)
+    expect(apiKey).toBeDefined()
+    expect(endpoint).toBeDefined()
+    GRAPHQL_CLIENT = new GraphQLClient(endpoint, { 'x-api-key': apiKey })
+});
+
+// afterAll(async () => {
```

Comment:
Can we un-comment these. Looks like we are not cleaning after ourselves 😄

Reply:
Yes good catch. Fixed.